### PR TITLE
#85 - refactor RockRoute(s) to the new service-repository alignment

### PR DESCRIPTION
**Pull request by gregito**

### CODE REVIEW

**Review by BenTrk — APPROVED** [review]
Nice refactor. I see I made you work pretty hard with the original code :D

**Comment by gregito** [comment]
> Nice refactor. I see I made you work pretty hard with the original code :D

Naah, it was fine 😄

File: lib/utils/log_utils.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  static void logEnd(int since, {String prefixMsg = 'Operation'}) {
+    double took = (DateTime.timestamp().millisecondsSinceEpoch - since) / 1000;
+    String msg = '$prefixMsg took ${took}s';
+    if (took < (debugThreshold as double)) {
+      logger.d(msg);
+    } else if (took >= (debugThreshold as double) &&
+        took < (infoThreshold as double)) {
+      logger.i(msg);
+    } else if (took >= (warnThreshold as double)) {}
```
Comment (terjekig) [inline lib/utils/log_utils.dart:25]: Isn't the "info < took < warning" case missing?
What would a 3.5sec time result in?

**Review by terjekig — CHANGES_REQUESTED** [review]

File: test/services/sector_service_test.dart [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+List<Route> createRoute({int quantity = 1}) {
+  List<Route> routes = [];
+  for (int i = 0; i < quantity; i++) {
+    routes.add(new Route(
+      length: quantity + 10,
+      key: quantity + 11,
+      difficulty: quantity,
+      diffchanger: quantity % 2 == 0 ? '+' : '-',
```
Comment (terjekig) [inline test/services/sector_service_test.dart:141]: nice :D

File: lib/utils/log_utils.dart (hunk 1 above)
Comment (gregito) [reply lib/utils/log_utils.dart]: @terjekig  updated

File: lib/utils/log_utils.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import 'package:kisgeri24/logging.dart';
+
+class LogUtils {
+  static const int debugThreshold = int.fromEnvironment(
+      'OPERATION_DURATION_DEBUG_THRESHOLD',
+      defaultValue: 2);
+  static const int infoThreshold =
+      int.fromEnvironment('OPERATION_DURATION_INFO_THRESHOLD', defaultValue: 3);
+  static const int warnThreshold =
+      int.fromEnvironment('OPERATION_DURATION_WARN_THRESHOLD', defaultValue: 5);
```
Comment (terjekig) [inline lib/utils/log_utils.dart:10]: Is this required in this case?

**Review by terjekig — CHANGES_REQUESTED** [review]
After update, warning treshold seems to be unused

File: lib/utils/log_utils.dart (hunk 3 above)
Comment (gregito) [reply lib/utils/log_utils.dart]: thanks, removed!

**Review by terjekig — APPROVED** [review]
Looks good, thank you! :)

**Pull request merged**